apps/froniusmonitor/fronius_monitor.star

```diff
@@ -0,0 +1,181 @@
+                fail("Fronius API request failed with status {}".format(rep.status_code))
+            data = rep.body()
+            cache.set(url, data, ttl_seconds = CACHE_TTL)
+        o = json.decode(data)
+    else:
+        o = DUMMY_DATA
+
+    data = o["Body"]["Data"]["Site"]
+    pv = data["P_PV"]
+    power_load = data["P_Load"]
+    grid = data["P_Grid"]
+
+    points = []
+    flows = []
+
+    img = SOLAR_PANEL if pv else SOLAR_PANEL_OFF
+    points.append((img, pv))
+    dir = 1 if pv else 0
+    flows.append(dir)
+
+    if power_load != None:
+        points.append((HOUSE, abs(power_load)))
+
+    if grid != None:
+        points.append((GRID, abs(grid)))
+        if grid > 0:
+            dir = -1
+        elif grid < 0:
+            dir = 1
+        else:
+            dir = 0
+        flows.append(dir)
+
+    columns = []
+    for p in points:
+        img, power = p
+        (formatted, unit) = format_power(power)
+        columns.append(
+            render.Column(
+                main_align = "space_between",
+                cross_align = "center",
+                children = [
+                    render.Image(src = img),
+                    render.Text(
+                        content = formatted,
+                        height = 8,
+                        font = "tb-8",
+                        color = "#ffd11a",
+                    ),
+                    render.Text(
+                        content = unit,
+                        height = 8,
+                        font = "tb-8",
+                        color = "#ffd11a",
+                    ),
+                ],
+            ),
+        )
+    dots = [render.Box(width = 18)]
+    for dir in flows:
+        if dir:
+            el = render.Image(src = DOTS_LTR if dir == 1 else DOTS_RTL)
+        else:
+            el = render.Box(width = 10)
+        dots.append(
+            render.Stack(children = [render.Box(width = 21), el]),
+        )
+    return render.Root(
+        child = render.Stack(
+            children = [
+                render.Row(
+                    expanded = True,
+                    main_align = "space_between",
+                    children = columns,
+                ),
+                render.Column(
+                    children = [
+                        render.Box(height = 8),
+                        render.Row(expanded = True, children = dots),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+def format_power(p):
+    if p:
+        if p < 1000:
+            return (humanize.float("###.", p), "W")
+        else:
+            return (humanize.float("#,###.##", p / 1000), "kW")
+    else:
+        return "0", "W"
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "inverter_ip",
+                name = "Inverter IP Address",
+                desc = "Local IP Address of the Fronius inverter",
```

Comment:
While Tidbyt devices run on your local WiFI, the apps themselves execute in our backend, which means they have no way to access local IPs. If that's the plan for this app, then I'm afraid it won't work as things currently stand.

Reply:
Drat! Okay, thanks for confirming. I'll look into going down the Solar Web route.